From a2d324c75057f0ddd66cbe5b5e419fddda213cc2 Mon Sep 17 00:00:00 2001
From: Ashok Sidipotu <ashok.sidipotu@collabora.com>
Date: Mon, 28 Feb 2022 06:46:19 +0530
Subject: [PATCH] systemd: changes to run wireplumber instances in containers

- remove the pipewire daemon dependencies so that
only wireplumber policy daemon can run in containers.
- remove pipewire user
---
 src/systemd/system/wireplumber@.service.in | 8 +++-----
 src/systemd/user/wireplumber@.service.in   | 5 ++---
 2 files changed, 5 insertions(+), 8 deletions(-)

 Inappropriate [configuration]

diff --git a/src/systemd/system/wireplumber@.service.in b/src/systemd/system/wireplumber@.service.in
index ba8eefe..e4b985b 100644
--- a/src/systemd/system/wireplumber@.service.in
+++ b/src/systemd/system/wireplumber@.service.in
@@ -5,8 +5,7 @@
 # wireplumber@policy.service loads policy.conf, etc.
 [Unit]
 Description=Multimedia Service Session Manager (%i)
-After=pipewire.service
-BindsTo=pipewire.service
+After=sockets.target
 Conflicts=pipewire-media-session.service
 
 [Service]
@@ -19,9 +18,8 @@ SystemCallFilter=@system-service
 Type=simple
 ExecStart=@WP_BINARY@ -c %i.conf
 Restart=on-failure
-User=pipewire
-Environment=PIPEWIRE_RUNTIME_DIR=%t/pipewire
+Environment=PIPEWIRE_RUNTIME_DIR=/var
 Environment=GIO_USE_VFS=local
 
 [Install]
-WantedBy=pipewire.service
+WantedBy=multi-user.target
diff --git a/src/systemd/user/wireplumber@.service.in b/src/systemd/user/wireplumber@.service.in
index 9dbceff..7e21194 100644
--- a/src/systemd/user/wireplumber@.service.in
+++ b/src/systemd/user/wireplumber@.service.in
@@ -5,8 +5,7 @@
 # wireplumber@policy.service loads policy.conf, etc.
 [Unit]
 Description=Multimedia Service Session Manager (%i)
-After=pipewire.service
-BindsTo=pipewire.service
+After=sockets.target
 Conflicts=pipewire-media-session.service
 
 [Service]
@@ -23,4 +22,4 @@ Slice=session.slice
 Environment=GIO_USE_VFS=local
 
 [Install]
-WantedBy=pipewire.service
+WantedBy=multi-user.target
-- 
2.35.1

